src/sys/windows/tty.rs

```diff
@@ -9,19 +9,35 @@ use ::sys::winapi::um::fileapi::{OPEN_EXISTING, CreateFileW};
 use ::sys::winapi::um::handleapi::{INVALID_HANDLE_VALUE, CloseHandle};
 use ::sys::winapi::um::processenv::GetStdHandle;
 use ::sys::winapi::um::winbase::{STD_INPUT_HANDLE, STD_OUTPUT_HANDLE};
-use ::sys::winapi::um::winnt::{LPWSTR, HANDLE, GENERIC_READ, GENERIC_WRITE, FILE_SHARE_READ};
+use ::sys::winapi::um::winnt::{LPWSTR, HANDLE, GENERIC_READ, GENERIC_WRITE, FILE_SHARE_READ, FILE_SHARE_WRITE};
 use ::sys::winapi::um::wincon::{
-    ENABLE_PROCESSED_OUTPUT, ENABLE_WRAP_AT_EOL_OUTPUT, ENABLE_LINE_INPUT,
-    ENABLE_PROCESSED_INPUT,  ENABLE_ECHO_INPUT,         ENABLE_VIRTUAL_TERMINAL_PROCESSING,
-    PeekConsoleInputW
+    ENABLE_PROCESSED_OUTPUT,       ENABLE_WRAP_AT_EOL_OUTPUT, ENABLE_LINE_INPUT,
+    ENABLE_PROCESSED_INPUT,        ENABLE_ECHO_INPUT,         ENABLE_VIRTUAL_TERMINAL_PROCESSING,
+    ENABLE_VIRTUAL_TERMINAL_INPUT, PeekConsoleInputW
 };
 
+/// Enum std stream type
+///
+/// This can be passed around to easily evaluate which type of stream (or stream data)
+/// should be returned by a callee.
+#[derive(Copy, Clone, PartialEq)]
+pub enum StdStream {
+    IN,
+    OUT,
+}
+
+/// No longer needed
+///
+/// Sources using this need to be replaced with WindowsConIn/WindowsConOut implenentations
```

Comment:
Before I can simply accept this statement at face value, we need to talk a little bit about the endgame for the API changes for Windows support. This PR looks like it's going to break examples, and I'd hate to allow a commit that regressed examples from a previously working state.

Because this is an ANSI terminal escape crate, the desired design is this:

1. We do NOT expose any APIs before it is guaranteed that ANSI escapes have been enabled. 
2. The way that @ticki has agreed we should approach this is to use a central `struct` as go-to place to expose Termion's API. Therefore, we need to make something like a `Termion` struct that will handle the ANSI feature toggling by itself, and then expose structs for `stdin` and `stdout` with the Windows implementations we have here being just that: implementations. 

I feel like at this point, it would probably be appropriate to begin enforcing the convention that when an application uses Termion, they should disuse `println!` macros and get used to writing something like this:

```
let termion = Termion::new(); // Panicks if more than 1 gets instantiated
{
    let mut stdout = termion::stdout().lock();
    writeln!(stdout, "some format here");
}
```

Reply:
> As far as breaking examples go, I thought that was the point of creating all of these various issues, so we can work through them one by one? This isn't supposed to be 'everything is tested and working pr'. It's supposed to be a pr for one step.

Not sure if I was clear now -- I mean regressing from what was already working. I know it's not reasonable to have every single one of the examples working as-is -- but there are several that already were working with at least `cmd.exe`.

I guess most of the reason I think this is because I know cleanup passes are hard -- I'm hoping to minimize the cleanup pass I'm sure we'll have down the road.